From e2f41626289cb0a3a16f1aafc0db3bee8ec54a2b Mon Sep 17 00:00:00 2001
From: j1nx <p.steenbergen@j1nx.nl>
Date: Mon, 20 Jan 2025 12:52:39 +0000
Subject: [PATCH 1/1] Add an explicit --basedir

---
 dracut.sh | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/dracut.sh b/dracut.sh
index 88b14f3e..b41f4e87 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -176,6 +176,8 @@ Creates initial ramdisk images for preloading modules
   --tmpdir [DIR]        Temporary directory to be used instead of default
                          ${TMPDIR:-/var/tmp}.
   -r, --sysroot [DIR]   Specify sysroot directory to collect files from.
+  -b, --basedir [DIR]   Specify basedir directory to use for dracut, by
+                         default it is {sysrootdir}/usr/lib/dracut.
   -l, --local           Local mode. Use modules from the current working
                          directory instead of the system-wide installed in
                          /usr/lib/dracut/modules.d.
@@ -386,7 +388,7 @@ rearrange_params() {
     TEMP=$(
         unset POSIXLY_CORRECT
         getopt \
-            -o "a:m:o:d:I:k:c:r:L:fvqlHhMNp" \
+            -o "a:m:o:d:I:k:c:r:L:b:fvqlHhMNp" \
             --long kver: \
             --long add: \
             --long force-add: \
@@ -414,6 +416,7 @@ rearrange_params() {
             --long add-confdir: \
             --long tmpdir: \
             --long sysroot: \
+            --long basedir: \
             --long stdlog: \
             --long compress: \
             --long squash-compressor: \
@@ -703,6 +706,11 @@ while :; do
             PARMS_TO_STORE+=" '$2'"
             shift
             ;;
+        -b | --basedir)
+            basedir_l="$2"
+            PARMS_TO_STORE+=" '$2'"
+            shift
+            ;;
         -L | --stdlog)
             stdloglvl_l="$2"
             PARMS_TO_STORE+=" '$2'"
@@ -905,6 +913,7 @@ while (($# > 0)); do
 done
 
 [[ $sysroot_l ]] && dracutsysrootdir="$sysroot_l"
+[[ $basedir_l ]] && dracutbasedir="$basedir_l"
 
 export LC_ALL=C
 export LANG=C
-- 
2.43.0

